doc/plugins.dox

```diff
@@ -170,75 +171,175 @@ Geany's website at http://www.geany.org/Support/BuildingOnWin32.
-provided by the plugin API.
 @code
-GeanyPlugin			*geany_plugin;
-GeanyData			*geany_data;
+#include <geanyplugin.h>
 @endcode
 
-Now you can go on and write your first lines for your new plugin. As mentioned before,
-you will need to implement and fill out a couple of functions/macros to make the plugin work.
-So let's start with PLUGIN_VERSION_CHECK().
+@note If you use autoconf then config.h must be included even before that as usual.
+
+Now you can go on and write your first lines for your new plugin. As mentioned before, you will
+need to implement a couple of functions. The first mandatory one is @a geany_load_module(). Geany
+uses the presence fo this function to identify a library as a plugin. When Geany scans the
+pre-defined and user-configured plugin directories, it will take a look at each shared library (or
+DLL on Windows) to see if it exports a @a geany_load_module() symbol. Files lacking these will be
+ignored. The second mandatory one is an initialization function that is only called when the plugin
+becomes actually enabled (by the user or at startup).
+
+@subsection register Registering a Plugin
+
+Geany will always invoke this geany_load_module(), regardless of whether the user activates your
+plugin. In fact its purpose to probe if the plugin should even be presented to the user. Therefore
+you must use this function to register your plugin. Geany will pass a pointer to a GeanyPlugin
+instance which acts as a unique handle to your plugin. Use this pointer for registering and later
+API calls. It won't change for the life time of the plugin. Registering the plugin consists of a
+number of steps:
+
+  1. Filling in GeanyPlugin::info with metadata that is shown to the user.
+    - @ref PluginInfo::name : The name of your plugin
+    - @ref PluginInfo::description : A brief description.
+    - @ref PluginInfo::version : The plugin's version.
+    - @ref PluginInfo::author : Your contact information, preferably in the form "Name <email>".
+    .
+    Filling in all of them is recommended to provide the best user experience, but only the name is
+    truly mandatory. Since all of the strings are shown to the user they should be human readable.
+  2. Filling in GeanyPlugin::funcs with function pointers that are called by Geany.
+    - @ref GeanyPluginFuncs::init : an initialization function
+    - @ref GeanyPluginFuncs::cleanup : a finalization function
+    - @ref GeanyPluginFuncs::configure : a function that provides configuration (optional)
+    - @ref GeanyPluginFuncs::help : a function that provides help (optional)
+    - @ref GeanyPluginFuncs::callbacks : a pointer to an array of PluginCallback (optional).
+    .
+    @a init and @a cleanup are mandatory, the other ones depend on how advanced your plugin is.
+    Furthermore, the @a init and @a cleanup pair is called when the user activates and
+    deactivates your plugin in the Plugin Manager, and on startup and exit. So use these to do
```

Comment:
Indeed they probably are not, but just using the [provided sentence](https://github.com/geany/geany/pull/469#commitcomment-12732245) makes it totally unambiguous.